k8srm-prototype/pkg/api/claim_types.go

```diff
@@ -53,66 +37,30 @@ type DeviceClassSpec struct {
 	// +required
 	DeviceType string `json:deviceType,omitempty`
 
-	// Driver specifies the driver that should handle this class of devices.
-	// When a DeviceClaim uses this class, only devices published by the
-	// specified driver will be considered.
-	// +optional
-	Driver *string `json:driver,omitempty`
-
 	// Constraints is a CEL expression that operates on device attributes,
 	// and must evaluate to true for a device to be considered. It will be
 	// ANDed with any Constraints field in the DeviceClaim using this class.
+	//
 	// +optional
 	Constraints *string `json:"constraints,omitempty"`
 
-	// Device claims may represent be satisfied by choosing multiple
-	// devices instead of just a single device. The min/max fields control
-	// whether we want a single device, or a set of devices to satisfy a
-	// claim.
-
-	// MinDeviceCount is the minimum number of devices that should be selected
-	// when satsifying a claim using this class. Default is 1.
-	// +optional
-	MinDeviceCount *int `json:"minDeviceCount,omitempty"`
-
-	// MaxDeviceCount is the maximum number of devices that should be selected
-	// when sastisfying a claim using this class. No maximum, by default.
-	// +optional
-	MaxDeviceCount *int `json:"maxDeviceCount,omitempty"`
-
-	// MatchAttributes allows specifying a constraint within a set of
-	// chosen devices, without having to explicitly specify the value of
-	// the constraint.  For example, this allows constraints like "all
-	// devices must be the same model", without having to specify the exact
-	// model. We may be able to use this for some basic topology
-	// constraints too, by representing the topology as device attributes.
-	//
-	// Currently, these are just strings. However, we could make them
-	// structs, and include required vs preferred matches. Required matches
-	// would fail if not met, where as preferred would lower the score if
-	// not met. We could even allow low/medium/high priority and adjust the
-	// score differently for each.
+	// AdminAccess indicates that this class provides administrative access
+	// to the devices. Claims using a class with AdminAccess are expected
+	// to be used for monitoring or other management services for a device.
+	// They ignore all ordinary claims to the device with respect to access
+	// modes and any resource allocations. Access to these classes must be
```

Comment:
what are the use cases for this behavior?
since this is dynamically allocated, if there are no other claims, this will trigger an allocation of the device ... what if the device can only be allocated once? subsequent allocations will fail despite resource allocation will still show one available?

Reply:
This allows the driver to provide access to a maintenance pod/containers without the device becoming unavailable for real workloads. Without this, the maintenance pod would have to be privileged on the node.

Reply:
but leaking this into the API seems a bit odd, specially if people will start to abuse this option to implement different workflows ... kind of remember me the force delete option for Pod setting that DeletionTimestamp to 0 ... and now we can't roll it back without breaking people 🤷 

I recognize that I don't understand why there is a need of maintenance pods, but should not this part of the driver? there is a readiness check the driver does

Reply:
https://github.com/kubernetes/enhancements/pull/4611 is how it could be done with `ResourceClaimParameters` and a quota type.